africanus/gridding/perleypolyhedron/kernels.py

```diff
@@ -0,0 +1,87 @@
+import numpy as np
+from scipy.special import jn
```

Comment:
scipy is an optional codex dependency so imports and functions that depend on these imports need to be guarded for cases where scipy is not installed. You can use the ``@requires_optional`` decorator here.

See https://codex-africanus.readthedocs.io/en/latest/util-api.html#africanus.util.requirements.requires_optional for info on how to use this.

```suggestion
try:
    from scipy.special import jn
except ImportError as e:
    opt_import_err = e
else:
    opt_import_err = None
```

The codex CI tests a base codex installation first, followed by a complete installation. The former will fail without the import guards.